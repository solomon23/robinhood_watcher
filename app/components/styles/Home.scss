
.equityContainer {
  text-align: center;
  overflow: hidden;
  color: #000;
  font-size: 14px;
  display: flex;
  height: 100vh;
  flex-flow: column;
}

.header {
  padding: 8px 0;
  background-color: #21ce99;
  flex: 0 1 auto;
}

.equity {
  font-size: 26px;
  color: #fafafa;
  margin-bottom: 8px;
}

.afterHours {
  margin-top: 4px;
  color: #222;
}

.refresh {
  width: 20px;
  height: 20px;
  position: absolute;
  left: 8px;
  top: 8px;
  background-size: 100% 100%;
  background-image: url('../../../resources/reload.png');

  &:hover {
    cursor: pointer;
  }
}

.reloading {
  animation: rotate 1.5s linear infinite;
}

@keyframes rotate {
  100% {
      transform: rotate(360deg);
  }
}
