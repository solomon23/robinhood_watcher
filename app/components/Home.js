// @flow
import React, { Component } from 'react'
import classnames from 'classnames'
import { calculateEquity } from '../services/portfolio'
import styles from './styles/Home.scss'
import { USD } from '../services/utils'
import StockList from './StockList'

type Props = {
  portfolio: object,
  positions: array,
  actions: object,
  getAllData: () => void,
  startRefreshTimer: () => void
}

export default class Home extends Component<Props> {
  props: Props

  constructor(context, props) {
    super(context, props)

    this.refresh = this.refresh.bind(this)
  }

  componentDidMount() {
    const { startRefreshTimer } = this.props

    // getAllData()
    startRefreshTimer()
  }

  refresh() {
    const { getAllData } = this.props

    getAllData()
  }

  render() {
    const { portfolio, positions, actions } = this.props

    if (!Object.keys(portfolio).length > 0) {
      return null
    }

    const {
      equity,
      equityDifference,
      afterHoursEquityDifference,
    } = calculateEquity(portfolio)

    const isReloading = actions.portfolioLoading || actions.positionsLoading

    return (
      <div className={styles.equityContainer}>
        <div className={styles.header}>
          <div onClick={this.refresh} className={classnames(styles.refresh, { [styles.reloading]: isReloading })} />
          <div className={styles.equity}>
            {USD(equity)}
          </div>
          <div className={styles.change}>
            {equityDifference} Today
          </div>
          {afterHoursEquityDifference && (
            <div className={styles.afterHours}>
              {afterHoursEquityDifference} After Hours
            </div>
          )}
        </div>
        <StockList stocks={positions} />
      </div>
    )
  }
}
