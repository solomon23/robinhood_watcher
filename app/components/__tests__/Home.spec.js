import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import Home from '../Home'

Enzyme.configure({ adapter: new Adapter() })

let actions = null

const defaultState = {
  portfolio: { extended_hours_equity: null,  equity: 100, adjusted_equity_previous_close: 101 },
  positions: [],
  actions: {},
}

describe('Home component', () => {
  beforeEach(() => {
    actions = { getAllData: jest.fn(), startRefreshTimer: jest.fn() }
  })

  it('should display for open hours', () => {
    const component = shallow(<Home {...defaultState} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
    expect(actions.startRefreshTimer).toHaveBeenCalled()
  })

  it('should display for after hours', () => {
    const state = {
      ...defaultState,
      portfolio: { extended_hours_equity: 102,  equity: 100, adjusted_equity_previous_close: 101 },
    }

    const component = shallow(<Home {...state} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should show loading', () => {
    const state = { ...defaultState, actions: { portfolioLoading: true } }
    const component = shallow(<Home {...state} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should fire refresh', () => {
    const component = shallow(<Home {...defaultState} {...actions} />)
    component.find('.refresh').simulate('click')
    expect(actions.getAllData).toHaveBeenCalled()
  })
})
