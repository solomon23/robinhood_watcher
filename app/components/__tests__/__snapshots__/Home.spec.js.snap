// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should display for after hours 1`] = `
<div
  className="equityContainer"
>
  <div
    className="header"
  >
    <div
      className="refresh"
      onClick={[Function]}
    />
    <div
      className="equity"
    >
      $102.00
    </div>
    <div
      className="change"
    >
      -$1.00 (-0.99%)
       Today
    </div>
    <div
      className="afterHours"
    >
      +$2.00 (+2.00%)
       After Hours
    </div>
  </div>
  <StockList
    stocks={Array []}
  />
</div>
`;

exports[`Home component should display for open hours 1`] = `
<div
  className="equityContainer"
>
  <div
    className="header"
  >
    <div
      className="refresh"
      onClick={[Function]}
    />
    <div
      className="equity"
    >
      $100.00
    </div>
    <div
      className="change"
    >
      -$1.00 (-0.99%)
       Today
    </div>
  </div>
  <StockList
    stocks={Array []}
  />
</div>
`;

exports[`Home component should show loading 1`] = `
<div
  className="equityContainer"
>
  <div
    className="header"
  >
    <div
      className="refresh reloading"
      onClick={[Function]}
    />
    <div
      className="equity"
    >
      $100.00
    </div>
    <div
      className="change"
    >
      -$1.00 (-0.99%)
       Today
    </div>
  </div>
  <StockList
    stocks={Array []}
  />
</div>
`;
